[T12-1] Expense Tracker

winsonhys opened the pull request:
@JasonChong96 @jonathantjm @jcjxwy @Snookerballs

parvathy-sudhir-pillai reviewed (commented):
A very good effort, team! PR is failing CI, Take a look at the error log on Travis.

Suggestions to improve:
* Change product scope in the developer guide.
* Remove unnecessary/dead links that refer to AB 4
* Remove info about the AB4 product and not yours in the developer guide.
* For the login use case, make sure there is a universal username and password that can be used during testing.

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,57 +833,235 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user who has a limited budget |view my expenses over a certain period of time |learn from my past endeavours and better manage my budget
+
+|`* * *` |user who wants to save money |separate my expenses into different categories| ee where am I spending more money on and where my expenses go and cut them accordingly
```
"See" where am I spending more money on and where my expenses go and cut them accordingly

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,57 +833,235 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+1. User adds expenditure record
+2. Once validated, ExpenseTracker shows success message and details of the expenditure
+3. If expenditure causes budget to be below a certain threshold, show a warning message to the user
```
This is an exception (happens only when the expense is past the budget? Rephrase, not very clear). Should be under extensions as one of the exceptional cases.

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,57 +833,235 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+1. User keys in command to delete a given expenditure.
+2. Once deleted successfully, expense tracker shows a success message.
+3. The deleted expenditure is deleted from the UI
```
System responses are in passive voice. Follow the same convention as for the user (active).
For eg: System removes expenditure from UI (GUI?)

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,57 +833,235 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+[none]
+* 2a. Expenditure does not exist.
+[none]
+** 2a1. Error message shown to the user.
```
System displays error message

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,57 +833,235 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+* 2a. Budget is already set
+[none]
+** 2a1. Expense tracker prompts user the previously set budget and asks if user wants to continue
+** 2a2. User confirmed that he/she wants to continue
```
What if the user doesn't want to continue?

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,57 +833,235 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+*MSS*
+
+1. User keys in the command to turn on notifications
+2. Expense tracker already has a budget set.
```
Can be mentioned as a precondition. Not directly part of the MSS

parvathy-sudhir-pillai commented on docs/UserGuide.adoc:
```diff
@@ -49,110 +45,139 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+=== Signing up a User : `signup`
+Signs up a new user to the Expense Tracker
+
+Format: `signup USERNAME`
```
Be reminded that while testing, you may need to provide a global username and password to not trouble the testers to create an account first

parvathy-sudhir-pillai commented on docs/UserGuide.adoc:
```diff
@@ -49,110 +45,139 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+Sets a cap on the expense tracker. Will warn you if you try
+to add an expense that will exceed your budget
+
+Format: setBudget MONEY_WITH_TWO_DECIMAL_PLACES
```
Is this for a day/week/month?

parvathy-sudhir-pillai reviewed (commented):
Overall: Good job, Team T12-1, on meeting all requirements for the v1.1 team component. However, some areas need more work. Hope you can tackle those areas in upcoming features.

* README.adoc
1. Change the name of your product from AB-4 to Expense Tracker
2. Remove the top navigation bar and link to LOs
3. Change the View on GitHub link to reflect your repo
![screen shot 2018-10-07 at 6 02 02 am](https://user-images.githubusercontent.com/9066342/46576224-72819480-c9f7-11e8-94fa-ebe47a8fcd55.png)

*  Contact Us adoc
1. Update the links and information
2. Update the contact

* Developer Guide adoc
1. Update Appendix B. Product Scope. Still refers to AB-4

* User Guide adoc
1. Update example command (point 3) under section 2 QuickStart
2. Update features command format
3. Command to turn on/off notifications?

parvathy-sudhir-pillai reviewed (commented):
### Feedback for v1.2

**Read Me**
* Include the UI before the product introduction.

**Contact Us**
* Update link to issue tracker, how to pull request, email address to contact.
* Remove unnecessary links/information.

**User Guide**
* Be precise and clear in descriptions
![t12_1](https://user-images.githubusercontent.com/9066342/47262456-de95e980-d51b-11e8-8f86-9396e41148a4.png)
* setRecurringFrequency resets your budget. But the name of the command feels like it refers to an expense which happens always after the specified time. For e.g. Food is a recurring expense for which you spend 10 SGD everyday. Is that what you mean by budget reset as well? If yes, this is not very clear. Rephrase the description to reflect the actual purpose of the command.

**Developer guide**
* Start changing the set up instructions
* Update sequence diagram for the logic component
* Include descriptions of Budget and Username in the class diagram for Model component.
Specify the name of the list of expenses as depicted in the CD.
![t12_2](https://user-images.githubusercontent.com/9066342/47262558-17cf5900-d51e-11e8-8b09-c18a9b8a4071.png)
* Update Section 3.6 Common Classes 
* Update the Implementation details for Undo/Redo feature. It still refers to AB.
* Include sequence diagrams for the implementation details of the different features. Keep in mind that we discourage low level diagrams that span multiple components. Try to draw diagrams at multiple levels: one that shows component level interactions and others shown low level interactions within one components.
* Remove 4.3. [Proposed] Data Encryption if you are not implementing
* Section 4.7.2 should be named Design Considerations to keep the style uniform
* Update Appendix A: Product Scope
* Use case steps preferably should be single operations
![t12_3](https://user-images.githubusercontent.com/9066342/47262631-8a413880-d520-11e8-8a87-9fcdb4d713ab.png)
* Use case log in has Use case ends on the same line as step 6a1

### Overall
* Good progress, Team! Particularly good effort on adding/updating implementation descriptions.
* Not enough new/updated UML diagrams to give feedback on. Please try to add more soon.
* Let's push the product towards a user-testable state by v1.3

JasonChong96 committed:
Fix checkstyle errors

JasonChong96 committed:
Created unit tests for LoginCommand and seperated from Integration tests

JasonChong96 committed:
Add unit tests for SignUpCommand and seperated them from Integration tests

Commit:
Merge pull request #173 from jonathantjm/develop

Fix StatsCommand and User Guide bugs

JasonChong96 committed:
Add unit tests for SetPasswordCommand and seperated it from Integration Tests

JasonChong96 committed:
Fix checkstyle errors

JasonChong96 committed:
Add decrypt and encrypt commands to ExpenseTrackerParser unit tests

JasonChong96 committed:
Remove redundant checked exception from ExpenseTrackerSystemTest#assertApplicationStartingStateIsCorrect

Snookerballs committed:
Fix Comment errors

Commit:
Change verbe to past tense

Commit:
Remove unique from test case name

Commit:
checkstyle

winsonhys committed:
Merge pull request #172 from JasonChong96/develop

Bug fixes based on mock practical and implement encrypt/decrypt commands

Snookerballs committed:
Implemented CategoryPanel and Tests

Snookerballs committed:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into CS2103-AY1819S1-T12-1-develop

# Conflicts:
#	src/main/java/seedu/address/logic/commands/SetCategoryBudgetCommand.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java

Snookerballs committed:
Fix Category not being able to accept single characters

Snookerballs committed:
Fixed Travis Error

Snookerballs committed:
Codacy Fix

Snookerballs committed:
Travis Fix

jcjxwy committed:
Merge pull request #16 from CS2103-AY1819S1-T12-1/develop

pr

winsonhys committed:
Merge pull request #175 from jcjxwy/master

Allow expense tracker to take in duplicate expenses. Debug for massedit command.

Snookerballs committed:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into CS2103-AY1819S1-T12-1-develop

# Conflicts:
#	src/test/java/seedu/address/model/ExpenseTrackerTest.java

Commit:
Merge pull request #170 from Snookerballs/develop

Fixed Tips and Implemented Category Panel

winsonhys committed:
Changed to warning notification

winsonhys committed:
Merged main

winsonhys committed:
Merged main

Commit:
Changes to user guide

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

jcjxwy committed:
Merge pull request #17 from CS2103-AY1819S1-T12-1/develop

pr

winsonhys committed:
Jaime's code

winsonhys committed:
Added setCategoryBudget on existing expense test

winsonhys committed:
Fixed XMlAdaptation of CategoryBudget

winsonhys committed:
Checkstyle

winsonhys committed:
Moved +

winsonhys committed:
Removed all debug statements

winsonhys committed:
Merged main

JasonChong96 committed:
Add userguide section for encrypt/decrypt and change app icon/title

JasonChong96 committed:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

Commit:
Update guides

JasonChong96 committed:
Remove redundant print statement

Commit:
Merge pull request #179 from JasonChong96/develop

Add userguide section for encrypt/decrypt and change app icon/title

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

# Conflicts:
#	docs/UserGuide.adoc

winsonhys committed:
Merged main

Commit:
Update user guide

winsonhys committed:
Adding an expense being overbudget now shows a notification instead of a warning (#176)

* Changed to warning notification

* Merged main

* Jaime's code

* Added setCategoryBudget on existing expense test

* Fixed XMlAdaptation of CategoryBudget

* Checkstyle

* Moved +

* Removed all debug statements

Snookerballs committed:
Merge pull request #18 from CS2103-AY1819S1-T12-1/develop

Pull

Snookerballs committed:
Fix Category Icons

Snookerballs committed:
Merge branch 'develop' of https://github.com/Snookerballs/main into develop

# Conflicts:
#	src/main/java/seedu/address/storage/JsonTipsStorage.java

Snookerballs committed:
Fixing Build Error

JasonChong96 committed:
Merge pull request #181 from Snookerballs/develop

Fix Build Error

Commit:
Update user guide typo

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

winsonhys committed:
Fixed spendings not getting restarted after reccurrence frequency

winsonhys committed:
Changed command message

winsonhys committed:
Merged main

winsonhys committed:
Removed debug statement

Commit:
Merge pull request #180 from jonathantjm/develop

Update user guide

winsonhys committed:
User doc update

winsonhys committed:
edited accoriding to comments

winsonhys committed:
Merge branch 'develop' into develop

winsonhys committed:
checkstyle

winsonhys committed:
merged main

Commit:
Fix image extensions

Commit:
Fix image file extensions

Commit:
Update user guide

winsonhys committed:
Now compares expense date before deleting

Commit:
Change Date implementation to use LocalDateTime

winsonhys committed:
Merge pull request #183 from jonathantjm/develop

Change Date implementation to use LocalDateTime. Update User guide

Commit:
Fix bug that default date does not store time

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

winsonhys committed:
Merge pull request #184 from jonathantjm/develop

Fix bug with constructing Date for current date

winsonhys committed:
User doc

winsonhys committed:
Merged main

winsonhys committed:
Now deletes properly

winsonhys committed:
Fixed tests

winsonhys committed:
Fixed tests

winsonhys committed:
Fixed Json tips bs, fixed category budget related stuff I HOPE (#182)

* Changed to warning notification

* Merged main

* Jaime's code

* Added setCategoryBudget on existing expense test

* Fixed XMlAdaptation of CategoryBudget

* Checkstyle

* Moved +

* Removed all debug statements

* Fixed spendings not getting restarted after reccurrence frequency

* Changed command message

* Removed debug statement

* User doc update

* edited accoriding to comments

* checkstyle

* Now compares expense date before deleting

* User doc

* Now deletes properly

* Fixed tests

* Fixed tests

Snookerballs committed:
Merge pull request #19 from CS2103-AY1819S1-T12-1/develop

Pull

Commit:
Update javadocs

Commit:
add link to code in ppp

jcjxwy committed:
Merge pull request #18 from CS2103-AY1819S1-T12-1/develop

pr

Commit:
Merge branch 'master' of https://github.com/jcjxwy/main

# Conflicts:
#	src/main/java/seedu/address/storage/JsonTipsStorage.java

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

Commit:
Refactor to improve code quality

Commit:
Add a logger to MassEditCommand

Commit:
Update developer guide

Commit:
Merge pull request #185 from jonathantjm/develop

Update documentation

Commit:
Checkstyle

Snookerballs committed:
Update category panel for certain commands

Commit:
Add test case to increase coverage

Snookerballs committed:
Fixing bug with BudgetCap

winsonhys committed:
Merged main

parvathy-sudhir-pillai reviewed (commented):
### Mid v1.4 Review
* Contact Us page: Issue tracker link leads to Error 404. Also, email to contact should be changed.
* Developer guide: Figures 1,2,3,13 still refer to AB-4 events and components. Please change this to match your apps. 
* Fig 13, 22, 18 shows lower level interactions between (Logic and Model), (Logic, Model and EventCenter). Use multiple SDs: one to show within component lower level interactions and another to show higher level interaction between components.
* Fig 21 show Logic as a rectangle, rather than an L. Otherwise, it may feel as a continuation of the Logic and Model component.
* CS2103 encourages multi-level sequence diagrams. Higher level interactions between components (e.g. Logic and Model) in a separate diagram and lower level interactions between sub-components (e.g. Parser and Command) within a component (Logic) in a separate diagram. The lower level diagram should not include interactions between components (e.g. Logic and Model).
* Please definitely redraw Fig 23. Separate out the Logic and Model into two diagrams. It is difficult for the reviewer to follow.
* The styles of Fig 25, 26 are better to follow for the rest of SDs.

**Good job, Team!
All the best for your v1.4 submission and presentation!**

winsonhys committed:
Documentation

Snookerballs committed:
Updated Documentation and Refactored Code
- Updated Developer Guide use cases, and user stories
- Updated PPP
- Updated User Guide, notifications section
-

winsonhys committed:
Fixed issue tracker

winsonhys committed:
Now cannot input too large numbers

Commit:
Merge pull request #186 from jcjxwy/master

Refactor to improve code quality

winsonhys committed:
Now cost cannot have too high values

Snookerballs committed:
Category Icon Fixes, Docs Update and Test implementation
- Implemented tests for XmlAdaptedNotificationHandler
- Fixed a bug with CategoryIcon where the percentage could be greater than 2dp
- Updated Dev Docs and User Guide

Snookerballs committed:
Checkstyle Fix

Snookerballs committed:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into CS2103-AY1819S1-T12-1-develop

# Conflicts:
#	docs/DeveloperGuide.adoc
#	src/main/java/seedu/address/logic/commands/MassEditCommand.java

Snookerballs committed:
Codacy Fix

jcjxwy committed:
Merge pull request #19 from CS2103-AY1819S1-T12-1/develop

pr

Commit:
Add Design Consideration for DG of find command

Commit:
Merge branch 'master' of https://github.com/jcjxwy/main

winsonhys committed:
merged main

Commit:
Minor refactoring for coding standards

winsonhys committed:
documentation

winsonhys committed:
Changed documentation

Snookerballs committed:
Fix Spellng/Grammar Mistakes.

Commit:
Fix developer guide references to address book

Commit:
Merge pull request #188 from winsonhys/develop

Added documentation. Now prevents people who are too rich from using our app.

Commit:
Merge pull request #189 from jonathantjm/develop

Refactoring and changes to diagrams in DG

Snookerballs committed:
Docs update

winsonhys committed:
Changed PNG to png

Snookerballs committed:
Travis Fix

Snookerballs committed:
Merge pull request #21 from CS2103-AY1819S1-T12-1/develop

Pull

Snookerballs committed:
Updating Docs

Commit:
Merge pull request #187 from Snookerballs/develop

Code Refactor, Bug Fixing for CategoryIcon, and Docs Update

Commit:
Refactors seedu.address to seedu.expensetracker

Commit:
Fix travis. More refactoring

Commit:
Merge pull request #190 from jonathantjm/refactor

Refactors seedu.address to seedu.expensetracker

Snookerballs committed:
Merge pull request #22 from CS2103-AY1819S1-T12-1/develop

Pull

Snookerballs committed:
Merge branch 'develop' of https://github.com/Snookerballs/main into develop

Commit:
Add description for Commons class in developer guide

JasonChong96 committed:
Update documentation and PPP

JasonChong96 committed:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

Commit:
Update PPP

Commit:
Adjust the size of image

JasonChong96 committed:
Resolve comments

JasonChong96 committed:
Add markup to String references

jcjxwy committed:
Merge pull request #20 from CS2103-AY1819S1-T12-1/develop

pr

Commit:
typo error

Commit:
Merge branch 'master' of https://github.com/jcjxwy/main

Commit:
Add markup

winsonhys committed:
docs

JasonChong96 committed:
Fix link to PPP

Commit:
Some modifications.

Commit:
Merge pull request #191 from JasonChong96/develop

Update documentation and PPP

Commit:
Updates to user guide

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

JasonChong96 committed:
Code refactoring

JasonChong96 committed:
Change setpassword to setPassword

Snookerballs committed:
Updated Docs
- Replace images with images of Expense Tracker with correct font
- Updated PPP , User Guide and Developer Guide

JasonChong96 committed:
Remove trailing whitespace

JasonChong96 committed:
Remove unused import

winsonhys committed:
documentation

JasonChong96 committed:
Restore support for select command

winsonhys committed:
Merged main

Snookerballs committed:
Changed UI Images, updated Docs

winsonhys committed:
10 page

Snookerballs committed:
Minor Comment Refactors

Commit:
Merge pull request #193 from JasonChong96/develop

Code refactoring and change command word to match camelCase

Commit:
resolve review

Commit:
Merge pull request #194 from Snookerballs/develop

Updated GUI Imaged and Docs

Commit:
Updates to documetnation

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

Commit:
Update readme

Commit:
Fix PPP and UG

JasonChong96 committed:
Shortened PPP to 10 pages

JasonChong96 committed:
Add evidence of logging

JasonChong96 committed:
Added more test cases for isValidName

JasonChong96 committed:
Add diagram for userguide

Commit:
Update tests

JasonChong96 committed:
Add no-whitespace criteria to SignUpCommand usage message

winsonhys committed:
Manual testing

winsonhys committed:
Merged main

Commit:
Merge pull request #195 from jonathantjm/develop

Small refactoring. Updates to docs

jcjxwy committed:
Merge pull request #21 from CS2103-AY1819S1-T12-1/develop

pr

Commit:
Resolve review

Snookerballs committed:
Fixed the status bar CSS styling.

Commit:
Merge pull request #192 from jcjxwy/master

Update PPP

Commit:
Merge pull request #197 from Snookerballs/develop

Fixed the status bar CSS styling and Docs Update

Commit:
typo error

JasonChong96 committed:
Add manual testing instructions

JasonChong96 committed:
Merge branch 'develop' into develop

Snookerballs committed:
Add Login extensions for use cases, updated manual testing

Commit:
Fix UG

Commit:
Add code snippet. Add stats manual testing

Snookerballs committed:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into CS2103-AY1819S1-T12-1-develop

# Conflicts:
#	docs/DeveloperGuide.adoc

Commit:
Merge pull request #199 from Snookerballs/develop

Add Login extensions for use cases, updated manual testing

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

# Conflicts:
#	docs/DeveloperGuide.adoc

Commit:
Update PPP

Commit:
Merge pull request #200 from jonathantjm/develop

Add stats user testing, add code snippet

JasonChong96 committed:
Update diagrams

Commit:
Update manual testing instruction

jcjxwy committed:
Merge pull request #22 from CS2103-AY1819S1-T12-1/develop

pr

JasonChong96 committed:
Merge pull request #198 from JasonChong96/develop

Update documentation

winsonhys committed:
docs

winsonhys committed:
merged main

jcjxwy committed:
Merge branch 'develop' into master

winsonhys committed:
Updated about me, changed to 'you'

Commit:
add prerequisites

Commit:
Merge branch 'master' of https://github.com/jcjxwy/main

winsonhys committed:
checkstyle

winsonhys committed:
Added numbering

Commit:
Merge pull request #196 from winsonhys/develop

PPP

jcjxwy committed:
Merge pull request #23 from CS2103-AY1819S1-T12-1/develop

pr

winsonhys committed:
Merge pull request #201 from jcjxwy/master

Update manual testing instruction

winsonhys committed:
Merged main

JasonChong96 committed:
Fix encrypt decrypt use case

JasonChong96 committed:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

JasonChong96 committed:
Fix heading level

JasonChong96 committed:
Merge pull request #202 from JasonChong96/develop

Fix Developer Guide

Commit:
typo

jcjxwy committed:
Merge pull request #24 from CS2103-AY1819S1-T12-1/develop

pr

Commit:
Merge branch 'master' of https://github.com/jcjxwy/main

winsonhys committed:
changed namme

Commit:
Merge pull request #204 from winsonhys/develop

Fixed image

JasonChong96 committed:
Fix documentation

Snookerballs committed:
Update Docs (#205)

* Doc Update

* Minor Fix

* Added Label for Image

JasonChong96 committed:
Merge pull request #206 from JasonChong96/develop

Fix documentation

jcjxwy committed:
Merge pull request #25 from CS2103-AY1819S1-T12-1/develop

pr

winsonhys committed:
Added instructions for linux error

Commit:
typo

winsonhys committed:
Merge pull request #207 from winsonhys/develop

Added instructions for linux error

JasonChong96 committed:
Change applicaton version to 1.4

JasonChong96 committed:
Merge pull request #209 from JasonChong96/develop

Change applicaton version to 1.4

Commit:
Fix DG

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

Snookerballs committed:
Doc Fixes. Added Note about Budget Panel Texts (#210)

* Doc Update

* Minor Fix

* Added Label for Image

* Minor fix on Manual Testing

* no message

Commit:
Merge pull request #211 from jonathantjm/develop

Fix DG

JasonChong96 committed:
Merge pull request #208 from jcjxwy/master

fix typo

jcjxwy committed:
Merge pull request #26 from CS2103-AY1819S1-T12-1/develop

pr

winsonhys committed:
Fixed accoridng to telegram

JasonChong96 committed:
Fix typo in appendix

JasonChong96 committed:
Merge pull request #212 from JasonChong96/develop

Fix typo in appendix

winsonhys committed:
Merge pull request #213 from winsonhys/develop

Fixed accoridng to telegram

Snookerballs committed:
Doc Fix (#214)

* Doc Update

* Minor Fix

* Added Label for Image

* Minor fix on Manual Testing

* no message

* Doc fixing

Commit:
Minor changes

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

Commit:
Merge pull request #215 from jonathantjm/develop

Changes in docs

Commit:
Changes to DG and PPP

Commit:
Fix + and line break

jcjxwy committed:
Merge pull request #27 from CS2103-AY1819S1-T12-1/develop

pr

Commit:
Update stats implementation

Commit:
Merge pull request #216 from jonathantjm/develop

Minor updates to docs

Commit:
Merge branch 'master' of https://github.com/jcjxwy/main

# Conflicts:
#	docs/DeveloperGuide.adoc

Snookerballs committed:
Doc Fix and PPP Fix (#218)

* Doc Update

* Minor Fix

* Added Label for Image

* Minor fix on Manual Testing

* no message

* Doc fixing

* Minor fixes and updates

Commit:
Update stats images

Commit:
fix one more

Commit:
Merge branch 'develop' of https://github.com/CS2103-AY1819S1-T12-1/main into develop

Commit:
Merge pull request #220 from jonathantjm/develop

Update stats images

jcjxwy committed:
Merge pull request #219 from jcjxwy/master

fix line break

Snookerballs committed:
Hopefully Final Doc fix (#221)

* Doc Update

* Minor Fix

* Added Label for Image

* Minor fix on Manual Testing

* no message

* Doc fixing

* Minor fixes and updates

* Hopefully the last fix

JasonChong96 committed:
Fix typo in DG

JasonChong96 committed:
Merge pull request #222 from JasonChong96/develop

Fix typo in DG

Commit:
Update PPP

Commit:
Merge pull request #223 from jonathantjm/develop

Update PPP

Snookerballs committed:
Doc Fix (#224)

* Doc Update

* Minor Fix

* Added Label for Image

* Minor fix on Manual Testing

* no message

* Doc fixing

* Minor fixes and updates

* Hopefully the last fix

* Remove Duplicate  test

* no message

* Spelling error
